polkadot/cli/src/lib.rs

```diff
@@ -190,23 +185,22 @@ pub fn run<I, T>(args: I) -> error::Result<()> where
 
 	config.database_path = db_path(&base_path).to_string_lossy().into();
 
-	let (mut genesis_storage, boot_nodes) = PresetConfig::from_spec(chain_spec)
-		.map(PresetConfig::deconstruct)
+	if matches.is_present("build-spec") {
+		info!("Building chain spec");
+		let spec = chain_spec.load().expect("Error loading chain spec");
+		let json = spec.to_json();
+		print!("{}", json);
+		return Ok(())
+	}
+
+	let (genesis_storage, boot_nodes) = chain_spec.load()
+		.map(service::ChainSpec::deconstruct)
 		.unwrap_or_else(|f| (Box::new(move || 
 			read_storage_json(&f)
 				.map(|s| { info!("{} storage items read from {}", s.len(), f); s })
 				.unwrap_or_else(|| panic!("Bad genesis state file: {}", f))
 		), vec![]));
 	
-	if matches.is_present("build-genesis") {
```

Comment:
can it still output the raw genesis? this is crucial.

Reply:
added this as `polkatod build-spec --raw`